vividus-extension-selenium/src/main/java/org/vividus/ui/screenshot/AbstractScreenshotParametersFactory.java

```diff
@@ -26,73 +26,123 @@
+
     private PropertyMappedCollection<C> screenshotConfigurations;
     private String shootingStrategy;
     private Map<IgnoreStrategy, Set<Locator>> ignoreStrategies;
 
-    protected Optional<C> getDefaultConfiguration()
+    @Override
+    public Optional<ScreenshotParameters> create()
     {
-        return screenshotConfigurations.getNullable(shootingStrategy);
+        return getDefaultConfiguration().map(this::createParameters);
     }
 
-    protected Optional<C> getScreenshotConfiguration(Optional<C> screenshotConfiguration, BinaryOperator<C> merger)
+    @Override
+    public P create(Optional<C> screenshotConfiguration, String sourceKey, Map<IgnoreStrategy, Set<Locator>> ignores)
     {
         Optional<C> defaultConfiguration = getDefaultConfiguration();
-        if (screenshotConfiguration.isEmpty())
+
+        return screenshotConfiguration.map(currentConfig ->
         {
-            return defaultConfiguration;
+            patchIgnores(sourceKey, currentConfig, ignores);
+
+            C configuration = defaultConfiguration
+                    .map(defaultConfig -> getConfigurationMerger().apply(currentConfig, defaultConfig))
+                    .orElse(currentConfig);
+            return createParameters(configuration);
+        }).orElseGet(() -> {
+            C configuration = defaultConfiguration.orElseGet(this::createScreenshotConfiguration);
+            return createParameters(configuration, ignores);
+        });
+    }
+
+    protected abstract C createScreenshotConfiguration();
+
+    protected abstract P createScreenshotParameters();
+
+    protected abstract BinaryOperator<C> getConfigurationMerger();
+
+    protected abstract void configure(C config, P parameters);
+
+    private void patchIgnores(String sourceKey, ScreenshotConfiguration screenshotConfiguration,
+            Map<IgnoreStrategy, Set<Locator>> ignores)
+    {
+        Set<Locator> elementsToIgnore = getIgnoresFromOneOf(screenshotConfiguration.getElementsToIgnore(), sourceKey,
+                ignores.get(IgnoreStrategy.ELEMENT));
+        screenshotConfiguration.setElementsToIgnore(elementsToIgnore);
+
+        Set<Locator> areasToIgnore = getIgnoresFromOneOf(screenshotConfiguration.getAreasToIgnore(), sourceKey,
+                ignores.get(IgnoreStrategy.AREA));
+        screenshotConfiguration.setAreasToIgnore(areasToIgnore);
+    }
+
+    private Set<Locator> getIgnoresFromOneOf(Set<Locator> configIgnores, String sourceKey, Set<Locator> source)
+    {
+        if (!source.isEmpty())
+        {
+            Validate.isTrue(configIgnores.isEmpty(), "The elements and areas to ignore must be passed "
+                    + "either through screenshot configuration or %s", sourceKey);
+            LOGGER.atWarn().addArgument(sourceKey).log("The passing of elements and areas to ignore through {}"
+                    + " is deprecated, please use screenshot configuration instead");
```

Comment:
I propose to mark something with deprecated annotations to keep it on the radars